src/components/EmailNotifications/VerifyEmailAddress.js

```diff
@@ -0,0 +1,258 @@
+                          <IconCross
+                            css={`
+                              color: ${theme.negative};
+                            `}
+                          />
+                        ) : (
+                          <IconCheck
+                            css={`
+                              opacity: 0;
+                            `}
+                          />
+                        )
+                      }
+                      adornmentPosition="end"
+                      type="email"
+                      wide
+                      onChange={handleEmailAddressChange}
+                      placeholder="email@address.com"
+                      onBlur={handleEmailAddressBlur}
+                    />
+                    {emailInvalid && (
+                      <p
+                        css={`
+                          color: ${theme.negative};
+                          ${textStyle('body4')};
+                          float: left;
+                          margin-top: ${0.5 * GU}px;
+                        `}
+                      >
+                        Please enter a valid email address.
+                      </p>
+                    )}
+                  </Field>
+                </div>
+                <LegalTermsAndPolicy
+                  termsAccepted={termsAccepted}
+                  setTermsAccepted={setTermsAccepted}
+                  theme={theme}
+                />
+                <div
+                  css={`
+                    display: flex;
+                    justify-content: space-between;
+                    flex-direction: ${compactMode ? 'column' : 'row'};
+                    width: 100%;
+                    margin-bottom: ${1.5 * GU}px;
+                    margin-top: ${3 * GU}px;
+                  `}
+                >
+                  <ActionButtons
+                    compactMode={compactMode}
+                    onClick={onDeleteEmail}
+                    width={buttonWidth}
+                  >
+                    Delete email
+                  </ActionButtons>
+                  <ActionButtons
+                    compactMode={compactMode}
+                    mode="strong"
+                    disabled={emailInvalid || !termsAccepted}
+                    onClick={handleSubscribe}
+                    width={buttonWidth}
+                  >
+                    Send verification email
+                  </ActionButtons>
+                </div>
+              </>
+            )
+          }
+          return (
+            <Button
+              css={`
+                margin-top: ${3 * GU}px;
+              `}
+              mode="strong"
+              onClick={handleReSendEmail}
+            >
+              Resend verification email
+            </Button>
+          )
+        })()}
+      </div>
+    </div>
+  )
+})
+
+function LegalTermsAndPolicy({ termsAccepted, setTermsAccepted, theme }) {
```

Comment:
Let's just use `useTheme()` in the component :).

Reply:
why do a call to a hook if we are using it on the parent? don't get the idea around this comment

Reply:
Primarily for API reasons—it feels very unnatural to pass down `theme` as it's not something the component should expose (and be controlled by) in its public API.

Using `useTheme()` (which is very, very cheap—we almost never have to worry about its performance) allows us to abstract this completely away from a user. They don't need to provide it, simplifying how the component is used as well as which dependencies are needed to render the component. Imagine a parent component using this, which doesn't need `useTheme()`. We'd force that component to both know, and use `useTheme()`, despite it not being relevant.

Moreover, exposing it as a prop acts as an open door for mistakes to happen in the future. If we passed a hardcoded object down, this component's theme wouldn't change with the rest of the components if the user changed the theme (e.g. changed the site to be in dark mode).